src/ecl/command.ts

```diff
@@ -134,4 +135,36 @@ export class ECLCommands {
             }
         }
     }
+
+    sign() {
+        if (vscode.window.activeTextEditor && sessionManager.session) {
+            const editor = vscode.window.activeTextEditor;
+            const eclText = editor.document.getText();
+            sessionManager.session.digitalKeys().then(async keys => {
+                if (keys.length === 0) {
+                    vscode.window.showWarningMessage(localize("Environment has no digital keys"));
+                } else {
+                    const key = await vscode.window.showQuickPick(keys, {
+                        placeHolder: localize("Select Digital Key")
+                    });
+                    if (key) {
+                        const passphrase = await vscode.window.showInputBox({
+                            prompt: localize("Enter passphrase for") + ` "${key}"`
```

Comment:
Will the user need to select the key and enter the passphrase every time they sign the code  or will it be cached?

Reply:
Everytime - but it is a picklist.

Reply:
(Also folks can switch environments easily... )